src/hooks/useSovereignVenue.ts

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from "react";
+
+import {
+  setSovereignVenue,
+  setSovereignVenueIsLoading,
+  setSovereignVenueError,
+} from "store/actions/SovereignVenue";
+
+import { fetchSovereignVenue } from "api/venue";
+
+import { AnyVenue } from "types/venues";
+import { ReactHook } from "types/utility";
+
+import { WithId } from "utils/id";
+import { sovereignVenueSelector } from "utils/selectors";
+import { tracePromise } from "utils/performance";
+
+import { useDispatch } from "./useDispatch";
+import { useSelector } from "./useSelector";
+
+export interface UseSovereignVenueProps {
+  venueId?: string;
+}
+
+export interface UseSovereignVenueData {
+  sovereignVenue?: WithId<AnyVenue>;
+  sovereignVenueId?: string;
+  isSovereignVenueLoading: boolean;
+  errorMsg?: string;
+}
+
+export const useSovereignVenue: ReactHook<
+  UseSovereignVenueProps,
+  UseSovereignVenueData
+> = ({ venueId }) => {
+  const dispatch = useDispatch();
+
+  const {
+    venue: sovereignVenue,
+    isLoading: isSovereignVenueLoading,
+    errorMsg,
+  } = useSelector(sovereignVenueSelector);
```

Comment:
Add , isEqual) to the end of this useSelector, otherwise it won’t memo correctly due to the deeply nested structure of venues

Reply:
Lodash’s isEqual

Reply:
@0xdevalias fixed in 98b0388